plugin/src/main/java/org/opensearch/ml/sdkclient/SdkClientModule.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.ml.sdkclient;
+
+import org.apache.http.HttpHost;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.opensearch.OpenSearchException;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.json.jackson.JacksonJsonpMapper;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.transport.rest_client.RestClientTransport;
+import org.opensearch.common.inject.AbstractModule;
+import org.opensearch.core.common.Strings;
+import org.opensearch.sdk.SdkClient;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+
+/**
+ * A module for binding this plugin's desired implementation of {@link SdkClient}.
+ */
+public class SdkClientModule extends AbstractModule {
+
+    public static final String REMOTE_METADATA_ENDPOINT = "REMOTE_METADATA_ENDPOINT";
+    public static final String REGION = "REGION";
+
+    private final String remoteMetadataEndpoint;
+    private final String region; // not using with RestClient
+
+    /**
+     * Instantiate this module using environment variables
+     */
+    public SdkClientModule() {
+        this(System.getenv(REMOTE_METADATA_ENDPOINT), System.getenv(REGION));
+    }
+
+    /**
+     * Instantiate this module specifying the endpoint and region. Package private for testing.
+     * @param remoteMetadataEndpoint The remote endpoint
+     * @param region The region
+     */
+    SdkClientModule(String remoteMetadataEndpoint, String region) {
+        this.remoteMetadataEndpoint = remoteMetadataEndpoint;
+        this.region = region;
+    }
+
+    @Override
+    protected void configure() {
+        boolean local = Strings.isNullOrEmpty(remoteMetadataEndpoint);
+        if (local) {
+            bind(SdkClient.class).to(LocalClusterIndicesClient.class);
```

Comment:
Should bind to instance of class instead of class?

Reply:
No, because the injector will just see `SdkClient` in the transport action, so that has to be the key.  They won't know which instance it's bound to. That's the whole purpose of this module.  We had this discussion [here](https://github.com/opensearch-project/ml-commons/pull/2430#discussion_r1597261968) and this module was in response to your suggestion.